File: fib/fib.hs
fib :: Int -> Integer
fib 0 =
    1
fib n =
    fib' n 1 0

fib' :: Int -> Integer -> Integer -> Integer
fib' 1 prev prevprev =
    prev+prevprev
fib' n prev prevprev =
    fib' (n-1) (prev+prevprev) prev

File: fib/fib.rb
def fib(n)
    if n == 0 then
        return 1
    end
    
    i    = 1
    curr = 1
    prev = 0

    (i .. n).each do
        tmp  = curr
        curr = curr + prev
        prev = tmp
    end
    
    return curr
end

puts fib 13

